src/coreclr/gc/gc.cpp

```diff
@@ -13527,6 +13466,60 @@ void gc_heap::distribute_free_regions()
                 }
             }
         }
+
+#ifdef MULTIPLE_HEAPS
+        if (balance != 0)
```

Comment:
I am wondering, do you mean `else if` here?

Reply:
No, making these sequential rather than if-else is what allows regions to be distributed after some (but not all) surplus is added to the decommit list.